packages/task/src/browser/task-service.ts

```diff
@@ -0,0 +1,173 @@
+            }
+        });
+
+        // notify user that task has finished
+        this.taskWatcher.onTaskExit((event: TaskExitedEvent) => {
+            const signal = event.signal;
+            if (!this.isEventForThisClient(event.ctx)) {
+                return;
+            }
+
+            if (event.code === 0) {  // normal process exit
+                let success = '';
+
+                // this finer breakdown will not work on Windows.
+                if (signal && signal !== '0') {
+                    if (signal === '1') {
+                        success = 'Terminal Hangup received - ';
+                    } else if (signal === '2') {
+                        success = 'User Interrupt received - ';
+                    } else if (signal === '15' || signal === 'SIGTERM') {
+                        success = 'Termination Interrupt received - ';
+                    } else {
+                        success = 'Interrupt received - ';
+                    }
+                } else {
+                    success = 'Success - ';
+                }
+
+                success += `Task ${event.taskId} has finished. exit code: ${event.code}, signal: ${event.signal}`;
+                this.messageService.info(success);
+            } else {  // abnormal process exit
+                this.messageService.error(`Error: Task ${event.taskId} failed. Exit code: ${event.code}, signal: ${event.signal}`);
+            }
+        });
+    }
+
+    /** returns an array of known task configuration labels */
+    public async getTasks(): Promise<string[]> {
+        return Promise.resolve(this.taskConfigurations.getTaskLabels());
+    }
+
+    /** Returns an array of running tasks 'TaskInfo' objects */
+    public getRunningTasks(): Promise<TaskInfo[]> {
+        return this.taskServer.getTasks(this.getContext());
+    }
+
+    /** runs a task, by task configuration label */
+    public async run(taskName: string): Promise<void> {
+        let taskInfo: TaskInfo;
+        const task = this.taskConfigurations.getTask(taskName);
+        if (!task) {
+            this.logger.error(`Can't get task launch configuration for label: ${taskName}`);
+            return;
+        }
+
+        try {
+            taskInfo = await this.taskServer.run(this.massageTaskConfiguration(task), this.getContext());
+        } catch (error) {
+            this.logger.error(`Error launching task '${taskName}': ${error}`);
+            this.messageService.error(`Error launching task '${taskName}': ${error}`);
+            return;
+        }
+
+        this.logger.debug(`Task created. task id: ${taskInfo.taskId}, OS ProcessId: ${taskInfo.osProcessId} `);
+
+        // open terminal widget if the task is based on a terminal process:
+        if (taskInfo.terminalId !== undefined) {
+            this.attachTerminal(taskInfo.terminalId, taskInfo.taskId);
+        }
+    }
+
+    public async attach(terminalId: number, taskId: number): Promise<void> {
```

Comment:
what's the difference between attach and attachTerminal ? seems the same and only attachTerminal is used ?

Reply:
Should we just rename attachTerminal attach?

Reply:
True - good catch - I followed your suggestion